File: spark/src/main/java/com/jason/spark/core/Spark09RDDSortBy.java
package com.jason.spark.core;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

import java.util.Arrays;

/**
 * @Description:
 * @author: 贾森
 * @date: 2024年10月25日 17:54
 */
public class Spark09RDDSortBy {
    public static void main(String[] args) {
        SparkConf sparkConf = new SparkConf().setMaster("local[*]").setAppName("Spark09RDDSortBy");
        JavaSparkContext sc = new JavaSparkContext(sparkConf);
        JavaRDD<Integer> parallelize = sc.parallelize(Arrays.asList(1, 2, 3, 4, 5, 6,5,4,3,3,3), 2);
        JavaRDD<Integer> javaRDD = parallelize.distinct();
        JavaRDD<Integer> sortByRDD = javaRDD.sortBy(new Function<Integer, Integer>() {
            @Override
            public Integer call(Integer integer) throws Exception {
                return integer;
            }
        }, true, 0);
        sortByRDD.collect().forEach(System.out::println);
        sc.stop();
    }
}

File: spark/src/main/scala/com/jason/spark/core/rdd/Spark09_RDDSort_By.scala
package com.jason.spark.core

import org.apache.spark.{SparkConf, SparkContext}

object Spark09_RDDSort_By {
  def main(args: Array[String]): Unit = {
    val sparkConf = new SparkConf().setMaster("local[*]").setAppName("Spark06_RDD_GroupBy")
    val sc = new SparkContext(sparkConf)
    val rdd = sc.makeRDD(List(1, 2, 3, 4, 5, 6, 5, 4, 3, 3, 3),2)
    val distinctedRDD = rdd.distinct()
    val sortedRDD = distinctedRDD.sortBy(v => v,true,0)
    sortedRDD.collect().foreach(println)
  }

}
